src/rust/bitbox02/src/util.rs

```diff
@@ -110,4 +102,31 @@ mod tests {
         assert_eq!(truncate_str("test", 5), "test");
         assert_eq!(truncate_str("test", 6), "test");
     }
+
+    #[test]
+    fn test_strlen_ptr() {
+        assert_eq!(unsafe { strlen_ptr(b"\0".as_ptr()) }, 0);
+        assert_eq!(unsafe { strlen_ptr(b"a\0".as_ptr()) }, 1);
+        assert_eq!(unsafe { strlen_ptr(b"abcdef\0".as_ptr()) }, 6);
+        assert_eq!(unsafe { strlen_ptr(b"abcdef\0defghji".as_ptr()) }, 6);
+    }
+
+    #[test]
+    fn test_str_from_null_terminated() {
+        assert_eq!(str_from_null_terminated(b"\0"), Ok(""));
+        assert_eq!(str_from_null_terminated(b"hello\0"), Ok("hello"));
+        assert_eq!(str_from_null_terminated(b"hello\0world"), Ok("hello"));
+        // valid utf8.
+        assert_eq!(
+            str_from_null_terminated(b"\xc3\xb6\xc3\xa4\xc3\xbc \xf0\x9f\x91\x8c\0world"),
+            Ok("öäü 👌")
```

Comment:
This is funny because we don't really support it in C, or do we? :)

Reply:
That's true :joy: